### ISSUE TITLE
traverse_commits missing commits

### ISSUE THREAD

**Issue by narvik01** [ishepard/pydriller#272]
**Describe the bug**
Since version 2.5, the traverse_commits method between two merge commits started giving incorrect results when the newly merged commits do not have the same parent. It seems to only return the from and to commits.

**To Reproduce**
To reproduce you can use this example:
```
from pydriller import Repository as PydrillerRepository

if __name__ == "__main__":
    repo = PydrillerRepository("https://bitbucket.org/ondrejh_nedbank/test_repo.git", 
                               from_commit='aa66bc567b8d29c74ad2480e6f704cb37fe6b3ed', 
                               to_commit='d08a28e8e53add463a2b462f8220fdd5ef01cfab')
    commits = repo.traverse_commits()
    for c in commits:
        print(f"{c.hash}")
```

with pydriller 2.5 output is 
> d08a28e8e53add463a2b462f8220fdd5ef01cfab
> aa66bc567b8d29c74ad2480e6f704cb37fe6b3ed

with pydriller 2.4.1:
> d08a28e8e53add463a2b462f8220fdd5ef01cfab
> 6eb34ac0569cc68de7bb68a95526c8a90289352c
> 163ee55a1bcd6d865aab6143080481f61e7f271f
> aa66bc567b8d29c74ad2480e6f704cb37fe6b3ed

which is correct:
![Screenshot 2023-06-13 at 22 17 48](https://github.com/ishepard/pydriller/assets/1526858/c221298b-2c1b-456f-8349-21f74551bee7)

**OS Version:**
MacOS, Alpine Linux

**Comment by ishepard** [comment]
From the example you shared, the new behaviour is correct. The old behaviour was wrong, and the reason I changed was exactly this. 

From aa do d0 there are no commits. The other 2 commits are in other branches, so they shouldn't show up.

**Comment by narvik01** [comment]
Technically the commits are in multiple branches, including the `test` branch where the merge commits are from. But I understand now that you can't reach `to_commit` traversing via the two in the middle which is the point of this method.

Is there a way to do this in Pydriller then? Something like
```
git --no-pager log aa66bc567b8d29c74ad2480e6f704cb37fe6b3ed ^d08a28e8e53add463a2b462f8220fdd5ef01cfab --oneline

aa66bc5 (HEAD -> test, origin/test) Merged in develop (pull request #9)
163ee55 (origin/develop) Merged in from_develop_with_test_2 (pull request #8)
6eb34ac (origin/from_develop_with_test_2, from_develop_with_test_2) eee
```
